docs/developer-docs/smart-contracts/compile.mdx

```diff
@@ -0,0 +1,73 @@
+---
+keywords: [beginner, tutorial, build]
+---
+
+import { MarkdownChipRow } from "/src/components/Chip/MarkdownChipRow";
+
+# Compile
+
+<MarkdownChipRow labels={["Beginner", "Tutorial"]} />
+
+## Overview
+
+After you have written the code for your project's canisters, you need to compile the code into a WebAssembly module before it can be deployed on ICP. To build your canister's Wasm module, you can use the `dfx build` command. The `dfx build` command looks for source code to compile for each canister configured under the `canisters` section in the `dfx.json` configuration file. It can be used to build a specific canister or all canisters defined in the project's `dfx.json` file.
+
+Before using the `dfx build` command, first you need to create the canister with the `dfx canister create <canister name>` command:
+
+```
+dfx canister create hello_world
+```
+
+Alternatively, you can create all canisters in a project with the `--all` flag:
+
+```
+dfx canister create --all
+```
+
+When this command is executed, the following steps happen:
+
+- The canister is created locally using the source code file listed in the project's `dfx.json` file for that canister name.
+
+- A cycles wallet canister is created on the local network for the user principal that executed the command if one does not exist. This wallet canister's ID is returned in the command line output.
+
+- The canister you created is assigned a canister ID, which is returned in the command line output.
+
+Before building your canister code using `dfx build`, first verify the location of your project's files and their file names. If you modify these settings, be sure to edit your project's `dfx.json` file to reflect the current intended configuration.
+
+Then, build your canister with the command:
+
+```
+dfx build
+```
+
+This command must be run from within the project's directory.
```

Comment:
Another suggestion is to link to the `dfx build` cli reference here: https://internetcomputer.org/docs/current/developer-docs/developer-tools/cli-tools/cli-reference/dfx-build.

There might be other options that are helpful to developers.

Reply:
added a link to the dfx build CLI reference